### PR TITLE
Update dev dependencies

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch

Resolves: #   
Change-type: major|minor|patch   
Depends-on: <url>   
See: <url>   

---
Please check the CONTRIBUTING.md file for relevant information and some
guidance. Keep in mind that the CLI is a cross-platform application that runs
on Windows, macOS and Linux. Tests will be automatically run by balena CI on
all three operating systems, but this will only help if you have added test
code that exercises the modified or added feature code.

Note that each commit message (currently only the first line) will be
automatically copied to the CHANGELOG.md file, so try writing it in a way
that describes the feature or fix for CLI users.

If there isn't a linked issue or if the linked issue doesn't quite match the
PR, please add a PR description to explain its purpose or the features that it
implements. Adding PR comments to blocks of code that aren't self explanatory
usually helps with the review process.

If the PR introduces security considerations or affects the development, build
or release process, please be sure to highlight this in the PR description.

Thank you very much for your contribution!

### CODE REVIEW

File: lib/utils/compose.js [hunk 1]
```diff
@@ -443,7 +443,10 @@ export class BuildProgressUI {
 		this._startTime = Date.now();
 	}
 
-	end(summary = null) {
+	/**
+	 * @param {Dictionary<string> | undefined} summary
```
Comment (Page-) [inline lib/utils/compose.js:447]: This is to make it match the typing it ends up getting assigned to, the typescript update recognized that they were different (presumably it was defaulting to `any` before)

File: lib/utils/docker.ts [hunk 2]
```diff
@@ -193,19 +193,7 @@ export async function createClient(
 	opts: dockerode.DockerOptions,
 ): Promise<dockerode> {
 	const Docker = await import('dockerode');
-	const docker = new Docker(opts);
-	const { modem } = docker;
-	// Workaround for a docker-modem 2.0.x bug where it sets a default
```
Comment (Page-) [inline lib/utils/docker.ts:198]: This was causing problems for typings with the @types/dockerode update and the issue it works around was fixed in https://github.com/apocas/docker-modem/commit/6c346d131f1d8d172909f5b4d517a6425d8d2282 and is definitely included as we're running 3.x

File: lib/utils/helpers.ts [hunk 3]
```diff
@@ -89,10 +89,10 @@ export async function sudo(
 	await executeWithPrivileges(command, stderr, isCLIcmd);
 }
 
-export function runCommand<T>(commandArgs: string[]): Promise<T> {
+export async function runCommand<T>(commandArgs: string[]): Promise<T> {
 	const { isSubcommand } =
 		require('../preparser') as typeof import('../preparser');
-	if (isSubcommand(commandArgs)) {
+	if (await isSubcommand(commandArgs)) {
```
Comment (Page-) [inline lib/utils/helpers.ts:95]: This was previously always truthy, the typescript update caught this bug but I wonder if the unconditional behavior might actually be necessary, @pdcastro ?

Comment (pdcastro) [reply lib/utils/helpers.ts:95]: Maybe, needs testing. Currently focusing on CLI v13. cc: @srlowe

Comment (Page-) [reply lib/utils/helpers.ts:95]: I checked and it's actually only used in 3 places and they're all sub commands, which explains why this was working before

**Comment by Page-** [comment]
@balena-ci rebase

**Comment by Page-** [comment]
@balena-ci I self-certify!

**Pull request merged**